mne/decoding/search_light.py

```diff
@@ -81,6 +81,12 @@ def fit(self, X, y):
         self : object
             Return self.
         """
+        if len(np.unique(y)) <= 2:
+            from sklearn.preprocessing import LabelEncoder
+             
+            le = LabelEncoder()
+            y = le.fit_transform(y)
```

Comment:
don't do this at fitting, it might confuse the users if they passed y=[1, 2] but didn't use roc.

Instead, apply it at scoring. If len(np.unique(y_true)) == 2, then roc_auc_score(le.fit_transform(y_true))